// mtgsdk-ts/out/IMagic.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Color;
(function (Color) {
    Color[Color["White"] = 0] = "White";
    Color[Color["Blue"] = 1] = "Blue";
    Color[Color["Black"] = 2] = "Black";
    Color[Color["Red"] = 3] = "Red";
    Color[Color["Green"] = 4] = "Green";
})(Color = exports.Color || (exports.Color = {}));
var ColorIdentity;
(function (ColorIdentity) {
    ColorIdentity[ColorIdentity["W"] = 0] = "W";
    ColorIdentity[ColorIdentity["U"] = 1] = "U";
    ColorIdentity[ColorIdentity["B"] = 2] = "B";
    ColorIdentity[ColorIdentity["R"] = 3] = "R";
    ColorIdentity[ColorIdentity["G"] = 4] = "G";
})(ColorIdentity = exports.ColorIdentity || (exports.ColorIdentity = {}));
var Rarity;
(function (Rarity) {
    Rarity[Rarity["Basic Land"] = 0] = "Basic Land";
    Rarity[Rarity["Common"] = 1] = "Common";
    Rarity[Rarity["Uncommon"] = 2] = "Uncommon";
    Rarity[Rarity["Mythic Rare"] = 3] = "Mythic Rare";
    Rarity[Rarity["Timeshifted"] = 4] = "Timeshifted";
    Rarity[Rarity["Masterpiece"] = 5] = "Masterpiece";
})(Rarity = exports.Rarity || (exports.Rarity = {}));
var Layout;
(function (Layout) {
    Layout[Layout["normal"] = 0] = "normal";
    Layout[Layout["split"] = 1] = "split";
    Layout[Layout["flip"] = 2] = "flip";
    Layout[Layout["double-faced"] = 3] = "double-faced";
    Layout[Layout["token"] = 4] = "token";
    Layout[Layout["plane"] = 5] = "plane";
    Layout[Layout["scheme"] = 6] = "scheme";
    Layout[Layout["phenomenon"] = 7] = "phenomenon";
    Layout[Layout["leveler"] = 8] = "leveler";
    Layout[Layout["vanguard"] = 9] = "vanguard";
})(Layout = exports.Layout || (exports.Layout = {}));
var Legality;
(function (Legality) {
    Legality[Legality["Legal"] = 0] = "Legal";
    Legality[Legality["Banned"] = 1] = "Banned";
    Legality[Legality["Restricted"] = 2] = "Restricted";
})(Legality = exports.Legality || (exports.Legality = {}));
var SetType;
(function (SetType) {
    SetType[SetType["core"] = 0] = "core";
    SetType[SetType["expansion"] = 1] = "expansion";
    SetType[SetType["reprint"] = 2] = "reprint";
    SetType[SetType["box"] = 3] = "box";
    SetType[SetType["un"] = 4] = "un";
    SetType[SetType["from the vault"] = 5] = "from the vault";
    SetType[SetType["premium deck"] = 6] = "premium deck";
    SetType[SetType["duel deck"] = 7] = "duel deck";
    SetType[SetType["starter"] = 8] = "starter";
    SetType[SetType["commander"] = 9] = "commander";
    SetType[SetType["planechase"] = 10] = "planechase";
    SetType[SetType["archenemy"] = 11] = "archenemy";
    SetType[SetType["promo"] = 12] = "promo";
    SetType[SetType["vanguard"] = 13] = "vanguard";
    SetType[SetType["masters"] = 14] = "masters";
})(SetType = exports.SetType || (exports.SetType = {}));
var BoosterCardType;
(function (BoosterCardType) {
    BoosterCardType[BoosterCardType["marketing"] = 0] = "marketing";
    BoosterCardType[BoosterCardType["land"] = 1] = "land";
    BoosterCardType[BoosterCardType["common"] = 2] = "common";
    BoosterCardType[BoosterCardType["uncommon"] = 3] = "uncommon";
    BoosterCardType[BoosterCardType["rare"] = 4] = "rare";
    BoosterCardType[BoosterCardType["mythic rare"] = 5] = "mythic rare";
})(BoosterCardType = exports.BoosterCardType || (exports.BoosterCardType = {}));
